lib/kernel-manager.coffee

```diff
@@ -149,9 +151,9 @@ module.exports = KernelManager =
 
         kernelSpec.grammarLanguage = grammarLanguage
 
-        ConfigManager.writeConfigFile (filepath, config, onlyConnect = false) =>
-            kernel = new Kernel kernelSpec, grammar, config, filepath, onlyConnect
+        customKernelConnectionPath = path.join atom.project.rootDirectories[0].path, 'hydrogen', 'connection.json'
 
+        finishKernelStartup = (kernel) =>
```

Comment:
The thing is that this code:

```CoffeeScript
            @_runningKernels[grammarLanguage] = kernel

            startupCode = Config.getJson('startupCode')[kernelSpec.display_name]
            if startupCode?
                console.log 'executing startup code'
                startupCode = startupCode + ' \n'
                kernel.execute startupCode

            onStarted?(kernel)
```

used to run inside this callback:

```CoffeeScript
        ConfigManager.writeConfigFile (filepath, config) =>
            kernel = new Kernel kernelSpec, grammar, config, filepath

            @_runningKernels[grammarLanguage] = kernel

            startupCode = Config.getJson('startupCode')[kernelSpec.display_name]
            if startupCode?
                console.log 'executing startup code'
                startupCode = startupCode + ' \n'
                kernel.execute startupCode

            onStarted?(kernel)
```

but now it should run either with a normal kernel (inside the callback) or with a custom kernel (called directly).

I tried first running it outside of both scopes after the `try-catch` with something like:

```CoffeeScript
            kernel = ConfigManager.writeConfigFile (filepath, config) =>
                kernel = new Kernel kernelSpec, grammar, config, filepath, onlyConnect=false
            

            @_runningKernels[grammarLanguage] = kernel
            startupCode = Config.getJson('startupCode')[kernelSpec.display_name]
            ...
```

but the value returned by `ConfigManager` is not the kernel but a `Server`, taken from the console:

```javascript
Server {domain: null, _events: Object, _eventsCount: 0, _maxListeners: undefined, _connections: 0…}
```

Digging in the final value (returns) of the callbacks I went through several callbacks in `config-manager.coffee`, `find-port.coffee` and thought that "debugging" that could lead to modifying several deep files unnecessarily while I could encapsulate that code in a function and just pass it.

Of course, there is the option of duplicating the code in both blocks, inside and outside the function, but I thought that would be a complete no-no.

Maybe I'm missing something and you can guide me...